src/net_processing.cpp

```diff
@@ -1278,11 +1291,7 @@ bool static ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv,
         UpdatePreferredDownload(pfrom, State(pfrom->GetId()));
         }
 
-        // Change version
         connman.PushMessage(pfrom, CNetMsgMaker(INIT_PROTO_VERSION).Make(NetMsgType::VERACK));
```

Comment:
Er, that doesn't work, it just reintroduces the problem that we're attempting to solve.

A mutex for a group of these attributes may be unavoidable. I'll poke at it some more.